### ISSUE TITLE
Voting: Add Multi-option support

### ISSUE THREAD

**Issue by izqui** [aragon/aragon-apps#231]
For Signalling to be useful and be able to use the Voting app we need to add support for more than 2 options at the contract level. This will also require frontend changes to display votes properly.

## Spec modification proposal

- Add a `_newSignalingVote(bytes metadata, uint8 options)` function protected by the ACL under the same or a new role. The reason for adding a new function is that multi-option votes **CANNOT** have a script to be executed, because of the cognitive complexity of having different scripts be executed depending on the outcome. And the scamming potential if a no vote can mean different things.

- Signalling votes cannot be executed, which means that a vote cannot be prematurely closed. They will be open for the entire `voteTime`.

- Metadata is an arbitrary byte array that can follow one of the following 3 formats, being the 3rd one the fallback:
  - An `aragon.js` contentURI, currently `ipfs:` and `http:` are the only supported providers. The client will then fetch the voting metadata JSON file at that location (data avail problems).
  - The metadata JSON payload directly (expensive).
  - An arbitrary string that doesn't fall under the above categories which is just displayed as the question.

The format of the metadata JSON could be something like:

```js
{
  "specId": "metadata1.voting.aragonpm.eth",
  "metadata": {
     "question": "when binance?", // required
     "description": "We want to pay **$1m** for the listing", // optional, markdown formatted blob
     "descriptionSource" : "github:issue://aragon/governance#181", // overwrites description when loaded correctly
     "tags": ["exchange-talk", "extortion"],
     "options": ["Yes", "Meh", "Over my dead body"] // optional
  }
}
```

If the options array or an particular option is missing then label options as 'Option 1' to 'Option n'

Tagging @lkngtn and @luisivan for feedback and assigning @chris-remus as the owner of the issue for now.

**Comment by onbjerg** [comment]
We should make this change for metadata on binary votes, too.

Could we generalize this (instead of having a seperate function for signalling votes) in a way such that EVM scripts are only executed if # options is 2? I.e. `newVote` now has the signature `newVote(bytes metadata, uint8 options, bytes script)`, and `forward` simply calls `newVote(YES_NO_METADATA, 2, _script)`.

For this, we should also make `question` optional, where the absence of `question` implies that `script` is set and the path is rendered instead.

**Comment by bingen** [comment]
We should consider separating Binary and Multi-option Voting contracts.

**Comment by sohkai** [comment]
Closing as we've merged Survey (#287).
